builder/internals.go

```diff
@@ -540,9 +540,13 @@ func (b *Builder) create() (*daemon.Container, error) {
 	b.TmpContainers[c.ID] = struct{}{}
 	fmt.Fprintf(b.OutStream, " ---> Running in %s\n", utils.TruncateID(c.ID))
 
-	// override the entry point that may have been picked up from the base image
-	c.Path = config.Cmd[0]
-	c.Args = config.Cmd[1:]
+	if config.Cmd != nil {
```

Comment:
Sorry if irrelevant, I'm not familiar with this part of the code. It seems to me that `config.Cmd` is the result of a `runconfig.Parse` which happen to give `config.Cmd` a default value of `nil`. However, if for some reason this behavior changed to have a default value of `[]string{}`, the code inside the if would crash for the same reason as today.

All this to say that I'm in favor of testing the length of the `Cmd` array rather than `nil`:

    if len(config.Cmd) > 0 {